src/Module/Contacts.php

```diff
@@ -0,0 +1,1137 @@
+			'edit_hover' => L10n::t('Edit contact'),
+			'photo_menu' => Contact::photoMenu($rr),
+			'id' => $rr['id'],
+			'alt_text' => $alt_text,
+			'dir_icon' => $dir_icon,
+			'thumb' => ProxyUtils::proxifyUrl($rr['thumb'], false, ProxyUtils::SIZE_THUMB),
+			'name' => htmlentities($rr['name']),
+			'username' => htmlentities($rr['name']),
+			'account_type' => Contact::getAccountType($rr),
+			'sparkle' => $sparkle,
+			'itemurl' => (($rr['addr'] != "") ? $rr['addr'] : $rr['url']),
+			'url' => $url,
+			'network' => ContactSelector::networkToName($rr['network'], $rr['url']),
+			'nick' => htmlentities($rr['nick']),
+		];
+	}
+
+	/**
+	 * @brief Gives a array with actions which can performed to a given contact
+	 *
+	 * This includes actions like e.g. 'block', 'hide', 'archive', 'delete' and others
+	 *
+	 * @param array $contact Data about the Contact
+	 * @return array with contact related actions
+	 */
+	private static function contact_actions($contact)
+	{
+		$poll_enabled = in_array($contact['network'], [Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::OSTATUS, Protocol::FEED, Protocol::MAIL]);
+		$contact_actions = [];
+
+		// Provide friend suggestion only for Friendica contacts
+		if ($contact['network'] === Protocol::DFRN) {
+			$contact_actions['suggest'] = [
+				'label' => L10n::t('Suggest friends'),
+				'url'   => 'fsuggest/' . $contact['id'],
+				'title' => '',
+				'sel'   => '',
+				'id'    => 'suggest',
+			];
+		}
+
+		if ($poll_enabled) {
+			$contact_actions['update'] = [
+				'label' => L10n::t('Update now'),
+				'url'   => 'contacts/' . $contact['id'] . '/update',
+				'title' => '',
+				'sel'   => '',
+				'id'    => 'update',
+			];
+		}
+
+		$contact_actions['block'] = [
+			'label' => (intval($contact['blocked']) ? L10n::t('Unblock') : L10n::t('Block')),
+			'url'   => 'contacts/' . $contact['id'] . '/block',
+			'title' => L10n::t('Toggle Blocked status'),
+			'sel'   => (intval($contact['blocked']) ? 'active' : ''),
+			'id'    => 'toggle-block',
+		];
+
+		$contact_actions['ignore'] = [
+			'label' => (intval($contact['readonly']) ? L10n::t('Unignore') : L10n::t('Ignore')),
+			'url'   => 'contacts/' . $contact['id'] . '/ignore',
+			'title' => L10n::t('Toggle Ignored status'),
+			'sel'   => (intval($contact['readonly']) ? 'active' : ''),
+			'id'    => 'toggle-ignore',
+		];
+
+		if ($contact['uid'] != 0) {
+			$contact_actions['archive'] = [
+				'label' => (intval($contact['archive']) ? L10n::t('Unarchive') : L10n::t('Archive')),
+				'url'   => 'contacts/' . $contact['id'] . '/archive',
+				'title' => L10n::t('Toggle Archive status'),
+				'sel'   => (intval($contact['archive']) ? 'active' : ''),
+				'id'    => 'toggle-archive',
+			];
+
+			$contact_actions['delete'] = [
+				'label' => L10n::t('Delete'),
+				'url'   => 'contacts/' . $contact['id'] . '/drop',
+				'title' => L10n::t('Delete contact'),
+				'sel'   => '',
+				'id'    => 'delete',
+			];
+		}
+
+		return $contact_actions;
+	}
+
+}
```

Comment:
According to the PSR-2 coding standards, there should be a new line at the end of each file, please add one.